File: treino/pilha.java
import java.util.Scanner;

public class pilha{
	public static void main(String[] args){
		Scanner sc = new Scanner(System.in);

		int [] pilha = new int[1000];
		int topo = -1;

		while(sc.hasNext()){
			String comando = sc.next();
			if(comando.charAt(0) == 'P' && comando.charAt(1) == 'U'){
				int numero = sc.nextInt();
				topo ++;
				pilha[topo] = numero;
			}else if(comando.charAt(0) == 'P' && comando.charAt(1) == 'O'){
				topo --;
			}else{
				if(topo == -1){
					System.out.println("Pilha vazia");
				}else{
					for(int j = topo; j >= 0; j--){
						System.out.println(pilha[j] + " ");
					}
				}
			}
		}
	}
}

File: treino/pilha.c
#include <stdio.h>

int main(){
	char pilha[1000][100];// 1000 strings com ate 100 letras
	int topo = -1;

	char comando[50];

	while(scanf("%s", comando) != EOF){
		if(comando[0] == 'A'){
			topo ++;
			scanf("%s", pilha[topo]);
		}else if(comando[0] == 'U'){
			if(topo > 0){
				topo --;
			}
		}else{
			for(int i = topo; i >=0; i --){
				printf("%s\n", pilha[i]);
			}
		}
	}
	return 0;
}
